### PR TITLE
rds_cluster: split up the test target

### PR DESCRIPTION
**Pull request by goneri**
Depends-On: https://github.com/ansible-collections/amazon.aws/pull/941

Split up the test target to avoid the parallel execution.

### CODE REVIEW

File: tests/integration/targets/rds_cluster_promote/aliases [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# TODO: Cannot be tested in the CI because:
+# An error occurred (InvalidParameterValue) when calling the CreateDBCluster operation: Replication from cluster in s
```
Comment (alinabuzachis) [inline tests/integration/targets/rds_cluster_promote/aliases:2]:
```suggestion
# An error occurred (InvalidParameterValue) when calling the CreateDBCluster operation: Replication from cluster in same region is not supported
```

File: tests/integration/targets/rds_cluster_modify/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# defaults file for rds_cluster
+
+# Create cluster
+cluster_id: ansible-test-{{ inventory_hostname | replace('_','-') }}{{ tiny_prefix
```
Comment (alinabuzachis) [inline tests/integration/targets/rds_cluster_modify/defaults/main.yml:4]: I guess `inventory_hostname` must be removed.

File: tests/integration/targets/rds_cluster_tag/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+subnets:
+- {cidr: '10.{{ 256 | random(seed=tiny_prefix) }}.1.0/24', zone: '{{ aws_region }}a'}
+- {cidr: '10.{{ 256 | random(seed=tiny_prefix) }}.2.0/24', zone: '{{ aws_region }}b'}
+- {cidr: '10.{{ 256 | random(seed=tiny_prefix) }}.3.0/24', zone: '{{ aws_region }}c'}
+- {cidr: '10.{{ 256 | random(seed=tiny_prefix) }}.4.0/24', zone: '{{ aws_region }}d'}
+
+security_groups:
+- '{{ tiny_prefix }}-sg-1'
+- '{{ tiny_prefix }}-sg-2'
+- '{{ tiny_prefix }}-sg-3'
```
Comment (alinabuzachis) [inline tests/integration/targets/rds_cluster_tag/defaults/main.yml:37]: I guess we can delete those from this role.
```suggestion

```

File: tests/integration/targets/rds_cluster_restore/defaults/main.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+subnets:
+- {cidr: '10.{{ 256 | random(seed=tiny_prefix) }}.1.0/24', zone: '{{ aws_region }}a'}
+- {cidr: '10.{{ 256 | random(seed=tiny_prefix) }}.2.0/24', zone: '{{ aws_region }}b'}
+- {cidr: '10.{{ 256 | random(seed=tiny_prefix) }}.3.0/24', zone: '{{ aws_region }}c'}
+- {cidr: '10.{{ 256 | random(seed=tiny_prefix) }}.4.0/24', zone: '{{ aws_region }}d'}
+
+security_groups:
+- '{{ tiny_prefix }}-sg-1'
+- '{{ tiny_prefix }}-sg-2'
+- '{{ tiny_prefix }}-sg-3'
```
Comment (alinabuzachis) [inline tests/integration/targets/rds_cluster_restore/defaults/main.yml:37]: Probably these ones too.
```suggestion
```

File: tests/integration/targets/rds_cluster_restore/defaults/main.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+port: 3306
+tags_create:
+  Name: ansible-test-cluster-{{ tiny_prefix }}
+  Created_By: Ansible_rds_cluster_integration_test
+
+# Modify cluster
+new_cluster_id: ansible-test-cluster-{{ tiny_prefix }}-new
+new_port: 1155
+new_password: test-rds_password-new
+new_db_parameter_group_name: ansible-test-db-parameter-group-{{ tiny_prefix }}-new
```
Comment (alinabuzachis) [inline tests/integration/targets/rds_cluster_restore/defaults/main.yml:18]:
```suggestion

```

File: tests/integration/targets/rds_cluster_restore/defaults/main.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
+# defaults file for rds_cluster
+
+# Create cluster
+cluster_id: ansible-test-{{ inventory_hostname | replace('_','-') }}{{ tiny_prefix
+  }}
+username: testrdsusername
+password: test-rds_password
```
Comment (alinabuzachis) [inline tests/integration/targets/rds_cluster_restore/defaults/main.yml:7]:
```suggestion
password: "{{ lookup('password', 'dev/null length=12 chars=ascii_letters,digits') }}"
```

File: tests/integration/targets/rds_cluster_promote/defaults/main.yml [hunk 7]
```diff
@@ -0,0 +1,37 @@
+# defaults file for rds_cluster
+
+# Create cluster
+cluster_id: ansible-test-{{ inventory_hostname | replace('_','-') }}{{ tiny_prefix
+  }}
+username: testrdsusername
+password: test-rds_password
```
Comment (alinabuzachis) [inline tests/integration/targets/rds_cluster_promote/defaults/main.yml:7]:
```suggestion
password: "{{ lookup('password', 'dev/null length=12 chars=ascii_letters,digits') }}"```

File: tests/integration/targets/rds_cluster_modify/defaults/main.yml [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+  }}
+username: testrdsusername
+password: test-rds_password
```
Comment (alinabuzachis) [inline tests/integration/targets/rds_cluster_modify/defaults/main.yml:7]:
```suggestion
password: "{{ lookup('password', 'dev/null length=12 chars=ascii_letters,digits') }}"
```

File: tests/integration/targets/rds_cluster_create_sgs/defaults/main.yml [hunk 9]
```diff
@@ -0,0 +1,37 @@
+# defaults file for rds_cluster
+
+# Create cluster
+cluster_id: ansible-test-{{ inventory_hostname | replace('_','-') }}{{ tiny_prefix
+  }}
+username: testrdsusername
+password: test-rds_password
```
Comment (alinabuzachis) [inline tests/integration/targets/rds_cluster_create_sgs/defaults/main.yml:7]:
```suggestion
password: "{{ lookup('password', 'dev/null length=12 chars=ascii_letters,digits') }}"
```

File: tests/integration/targets/rds_cluster_create_sgs/defaults/main.yml [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+# Modify cluster
+new_cluster_id: ansible-test-cluster-{{ tiny_prefix }}-new
+new_port: 1155
+new_password: test-rds_password-new
+new_db_parameter_group_name: ansible-test-db-parameter-group-{{ tiny_prefix }}-new
+
+# Tag cluster
+tags_patch:
+  Name: '{{ tiny_prefix }}-new'
+  Created_by: Ansible rds_cluster integration tests
```
Comment (alinabuzachis) [inline tests/integration/targets/rds_cluster_create_sgs/defaults/main.yml:23]: Probably these ones aren't used in this role.
```suggestion

```

File: tests/integration/targets/rds_cluster_modify/defaults/main.yml [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+subnets:
+- {cidr: '10.{{ 256 | random(seed=tiny_prefix) }}.1.0/24', zone: '{{ aws_region }}a'}
+- {cidr: '10.{{ 256 | random(seed=tiny_prefix) }}.2.0/24', zone: '{{ aws_region }}b'}
+- {cidr: '10.{{ 256 | random(seed=tiny_prefix) }}.3.0/24', zone: '{{ aws_region }}c'}
+- {cidr: '10.{{ 256 | random(seed=tiny_prefix) }}.4.0/24', zone: '{{ aws_region }}d'}
+
+security_groups:
+- '{{ tiny_prefix }}-sg-1'
+- '{{ tiny_prefix }}-sg-2'
+- '{{ tiny_prefix }}-sg-3'
```
Comment (alinabuzachis) [inline tests/integration/targets/rds_cluster_modify/defaults/main.yml:37]: Probably not needed here.
```suggestion
```

File: tests/integration/targets/rds_cluster_restore/defaults/main.yml (hunk 6 above)
Comment (goneri) [reply tests/integration/targets/rds_cluster_restore/defaults/main.yml:7]: This is a code reorganization, I don't think this change is required at this stage. The current static password works as expected and if we were to auto-generate it, I would prefer to do it in another patch.

**Comment by goneri** [comment]
recheck

**Comment by goneri** [comment]
regate

**Comment by goneri** [comment]
regate

**Pull request merged**